src/Workspaces/Core/Portable/SolutionCrawler/InvocationReasons.cs

```diff
@@ -2,44 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System.Collections;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.Serialization;
 
 namespace Microsoft.CodeAnalysis.SolutionCrawler
 {
+    [DataContract]
     internal partial struct InvocationReasons : IEnumerable<string>
```

Comment:
>InvocationReasons [](http://example.com/codeflow?start=28&length=17)

Can this simply be a flags enum?

Reply:
the goal here is not change anything, but to use the existing concepts that solution-crawler is already doing.

Reply:
In-proc we seem to have pre-allocated instances of this, but OOP deserialization will allocate a new one each call.

Reply:
ok.  will move to enum in followup :)